src/OpenSearch.Net/Connection/HttpConnection.cs

```diff
@@ -431,19 +432,17 @@ private static async Task SetContentAsync(HttpRequestMessage message, RequestDat
 			}
 		}
 
-		private static System.Net.Http.HttpMethod ConvertHttpMethod(HttpMethod httpMethod)
-		{
-			switch (httpMethod)
+		private static System.Net.Http.HttpMethod ConvertHttpMethod(HttpMethod httpMethod) =>
+			httpMethod switch
 			{
-				case HttpMethod.GET: return System.Net.Http.HttpMethod.Get;
-				case HttpMethod.POST: return System.Net.Http.HttpMethod.Post;
-				case HttpMethod.PUT: return System.Net.Http.HttpMethod.Put;
-				case HttpMethod.DELETE: return System.Net.Http.HttpMethod.Delete;
-				case HttpMethod.HEAD: return System.Net.Http.HttpMethod.Head;
-				default:
-					throw new ArgumentException("Invalid value for HttpMethod", nameof(httpMethod));
-			}
-		}
+				HttpMethod.GET => System.Net.Http.HttpMethod.Get,
+				HttpMethod.POST => System.Net.Http.HttpMethod.Post,
+				HttpMethod.PUT => System.Net.Http.HttpMethod.Put,
+				HttpMethod.DELETE => System.Net.Http.HttpMethod.Delete,
+				HttpMethod.HEAD => System.Net.Http.HttpMethod.Head,
+				HttpMethod.PATCH => Patch,
```

Comment:
I think if I delete this line the tests still pass because they use this list to iterate over the methods being checked. I think you should duplicate it in tests.

Reply:
The test iterates based on the declared values of the libraries HttpMethod enum, not based on this switch. Which covers the bug that was introduced, Patch was added to the enum, but this didn't have a case added and fell through to the default exception. The test would have automatically added the PATCH test case based on the enum, and then subsequently failed.

Reply:
Oh my bad, I see.